platform/windows/display_server_windows.cpp

```diff
@@ -815,9 +815,17 @@ void DisplayServerWindows::_mouse_update_mode() {
 
 void DisplayServerWindows::mouse_set_mode(MouseMode p_mode) {
 	ERR_FAIL_INDEX(p_mode, MouseMode::MOUSE_MODE_MAX);
-	if (p_mode == mouse_mode_base) {
+
+	if (mouse_mode_override_enabled) {
+		mouse_mode_base = p_mode;
+		// No need to update, as override is enabled.
+		return;
+	}
+	if (p_mode == mouse_mode_base && p_mode == mouse_mode) {
+		// No need to update, as it is currently set as the correct mode.
 		return;
 	}
+
 	mouse_mode_base = p_mode;
 	_mouse_update_mode();
```

Comment:
`_mouse_update_mode` already have the following check:

```c++
	MouseMode wanted_mouse_mode = mouse_mode_override_enabled
			? mouse_mode_override
			: mouse_mode_base;

	if (mouse_mode == wanted_mouse_mode) {
		return;
	}
```

And `mouse_mode` is only and always updated in `_mouse_update_mode`, so these checks seems redundant (on desktop platforms).

Reply:
You're right, forgot about the check in update. Will revert the desktop files.